### PR TITLE
[#338] Build summoner with GHC-8.8

### LINKED ISSUE #337: Support `generic-deriving-1.13`
From this Stackage issue:

* https://github.com/commercialhaskell/stackage/issues/4781

### LINKED ISSUE #338: Build summoner with GHC-8.8
TODO:
- [x] Update Travis config
- [x] Add `-Wmissing-deriving-strategies` to `.cabal` file ghc options

### PR DESCRIPTION
**Pull request by vrom911**
Resolves #338
Resolves #337 

## Checklist:

* Update CHANGELOG
    - [x] [summoner-cli/CHANGELOG](https://github.com/kowainik/summoner/blob/master/summoner-cli/CHANGELOG.md).
    - [x] [summoner-tui/CHANGELOG](https://github.com/kowainik/summoner/blob/master/summoner-tui/CHANGELOG.md).
- [x] Keep code style used in the changed files (see [style-guide](https://github.com/kowainik/org/blob/master/style-guide.md#haskell-style-guide) for more details).
- [x] Use the [`stylish-haskell` file](https://github.com/kowainik/summoner/blob/master/.stylish-haskell.yaml).
- [x] Update documentation (README, haddock) if required.
- [x] Create a new test project using `summoner` and check that the changes work as expected.

*Hint:* Add the `[ci skip]` text to the docs-only related commit's name, so no need to wait for CI to pass.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -18,11 +18,14 @@ matrix:
...
   - ghc: 8.4.4
   - ghc: 8.6.5
+  - ghc: 8.8.1
 
   # Stack, linux
 
   - ghc: 8.6.5
     env: STACK_YAML="$TRAVIS_BUILD_DIR/stack.yaml" DEPLOY=yes
+  - ghc: 8.8.1
+    env: STACK_YAML="$TRAVIS_BUILD_DIR/stack-nightly.yaml" DEPLOY=yes
```
Comment (chshersh) [inline .travis.yml]: Let's keep only one of the two deploys

File: stack-nightly.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+resolver: nightly-2019-10-31
+
+packages:
+  - summoner-cli/
+  - summoner-tui/
+
+extra-deps:
+  - base-noprelude-4.13.0.0
```
Comment (chshersh) [inline stack-nightly.yaml]: One more reason to use `mixins` approach with default prelude: `base-noprelude` is not going to be on Stackage unless we add it :disappointed:

File: stack-nightly.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  - brick-0.50
+  - generic-deriving-1.12.4
```
Comment (chshersh) [inline stack-nightly.yaml]: Will we use newer `generic-deriving` under separate PR? We have an issue for that. I see that `brick` and `relude` were updated. Can do this with `generic-deriving` as well... https://github.com/kowainik/summoner/issues/337
What do you think?

File: stack.yaml [hunk 4]
```diff
@@ -1,22 +1,8 @@
...
-
-  # for tomland
-  - megaparsec-7.0.5
-  - parser-combinators-1.1.0
-
-  # for brick
-  - config-ini-0.2.4.0
-  - data-clist-0.1.2.2
-  - text-zipper-0.10.1
-  - word-wrap-0.4.1
```
Comment (chshersh) [inline stack.yaml]: Nice cleanup!

File: summoner-cli/summoner.cabal [hunk 5]
```diff
@@ -106,17 +111,17 @@ library
                      , neat-interpolation ^>= 0.3.2.2
                      , optparse-applicative ^>= 0.15
                      , process ^>= 1.6.1.0
-                     , relude ^>= 0.5.0
-                     , shellmet ^>= 0.0.1
+                     , relude >= 0.5.0 && < 0.7
```
Comment (chshersh) [inline summoner-cli/summoner.cabal]: Shouldn't we always use `^>= 0.6.0.0` for relude? Like, is it okay to support multiple versions of `relude` like with `base` or should we always specify the only version to guard ourselves?

Comment (vrom911) [reply summoner-cli/summoner.cabal]: We are building using stack with both versions of relude and having no issues, I think that it is okay, unless it brings troubles.

**Pull request merged**